changes.d/6809.break.md

```diff
@@ -0,0 +1 @@
+Dropped support for Python 3.7-3.11. The minimum supported version is now 3.12.
```

Comment:
Not exactly a breaking change, since it doesn't affect workflows. But I feel like this is the most apt changelog category